test/unit/specs/util/settingsUtils.spec.js

```diff
@@ -0,0 +1,30 @@
+import { hashCode } from '@/util/settingsUtils';
+
+describe('settingsUtils.js', () => {
+	describe('hashCode', () => {
+
+		it('returns 0 if no param is passed', () => {
+			expect(hashCode()).toEqual(0);
+		});
+
+		it('returns 0 if an empty string is passed', () => {
+			expect(hashCode('')).toEqual(0);
+		});
+
+		it('converts a simple string into a 32bit integer', () => {
+			const simpleString = 'some string';
+			expect(hashCode(simpleString)).toEqual(1395333309);
+		});
+
+		it('converts a string including digits into a 32bit integer', () => {
+			const simpleString = 'some string 12345';
+			expect(hashCode(simpleString)).toEqual(-472388848);
```

Comment:
Should we maybe have hashCode return the absolute value so that we don't have to encode the negative sign as well? That will increase the chance of collision, but hopefully not by much.